test: thread : test run k_thread_resume on unsuspend thread

yingmingx opened the pull request:
If calling function k_thread_resume() when the thread is not suspend,
it takes no effect. This change improve coverage of function
k_thread_resume() in sched.c

Signed-off-by: Ying ming <mingx.ying@intel.com>

nashif commented on tests/kernel/threads/thread_apis/src/test_threads_suspend_resume.c:
```diff
@@ -27,6 +27,9 @@ static void threads_suspend_resume(int prio)
 				      create_prio, K_USER, K_NO_WAIT);
+	/* checkpoint: resume an unsuspended thread has no effect */
+	k_thread_resume(tid);
+	zassert_false(last_prio == create_prio, NULL);
```
how is this check related to k_thread_resume()?

yingmingx replied:
Thank you for you comment, you're right. This check has no effect here. I'll change my code.

nashif commented on tests/kernel/threads/thread_apis/src/test_threads_suspend_resume.c:
```diff
@@ -27,6 +27,18 @@ static void threads_suspend_resume(int prio)
+	/* checkpoint: resume an unsuspended thread has no effect */
+	k_thread_resume(tid);
+	k_msleep(100);
+	zassert_true(last_prio == create_prio, NULL);
```
still not sure how this assert on priorities is testing that resume has no effect on unsuspended threads.

yingmingx replied:
I added the line 30-34 in this existing function.  The main purpose of these code is to make a comparison between line 42-46. The line 36-37 is just to restore the enviroment.
If I changed the line 31 k_thread_resume to fucntion k_thread_suspend. Then the assert on line 33 would fail.
The new added code can also reflect k_thread_resume() and k_thread_suspend() are totally different function. For example, they are not like mouse buttons, but like light switches.

yingmingx replied:
@nashif  Could you take another look at it, thanks!

nashif replied:
I am still not convinced you are testing what you claim. The priority of the thread does not indicate anything about what happens when you resume a running thread.

yingmingx replied:
I got you point.  It have changed the code use kernel API to get thread state to judge zassert. 
I've created a new testcase. The new testcase must test in kernel mode because get thread state is not allowed in user mode.

Commit:
test: thread : test run k_thread_resume on unsuspend thread

If calling function k_thread_resume() when the thread is not suspend,
it takes no effect. This change improve coverage of function
k_thread_resume() in sched.c

Signed-off-by: Ying ming <mingx.ying@intel.com>

enjiamai reviewed (approved):
The newly adding test case is in order to verify k_thread_resume() only do things when the previous state of the thread was suspended. Looks fine for me.

yingmingx commented:
@nashif  The previous judging condition use thread priority is not proper.
I used k_thread_state_str() to assess the thread status before and after runing k_thread_resume().
